### PR TITLE
add box coder for detection

### LINKED ISSUE #4364: Add box coder for detection
**Is your feature request related to a problem? Please describe.**
Add box coder for detection

**Describe the solution you'd like**
Add box coder for detection

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Can-Zhao**
Signed-off-by: Can Zhao <canz@nvidia.com>

Fixes #4364 .

### Description
add box coder for detection

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/detection/utils/box_coder.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    CornerCornerModeTypeB,
+    StandardMode,
+    convert_box_mode,
+    convert_box_to_standard_mode,
+    is_valid_box_values,
+)
+from monai.utils.module import look_up_option
+
+
+# @torch.jit._script_if_tracing
```
Comment (Nic-Ma) [inline monai/apps/detection/utils/box_coder.py:71]: May I know the use case of this decorator?

Thanks.

File: monai/apps/detection/utils/box_coder.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+def encode_boxes(gt_boxes: Tensor, proposals: Tensor, weights: Tensor) -> Tensor:
+    """
+    Encode a set of proposals with respect to some reference ground truth (gt) boxes.
+    Assuming all boxes are with xyxy or xyzxyz mode.
```
Comment (Nic-Ma) [inline monai/apps/detection/utils/box_coder.py:75]: Maybe better to say ``StandardMode`` instead of xyxy or xyzxyz mode.

Thanks.

**Review by Nic-Ma — APPROVED** [review]

File: monai/apps/detection/utils/box_coder.py (hunk 1 above)
Comment (Can-Zhao) [reply monai/apps/detection/utils/box_coder.py:71]: I don't know... It comes from torchvision. Seems to be some accelerating trick

Comment (Nic-Ma) [reply monai/apps/detection/utils/box_coder.py:71]: Please remove it if not necessary.
CC @wyli .

Thanks.

Comment (Can-Zhao) [reply monai/apps/detection/utils/box_coder.py]: Sure

**Pull request merged**